app/src/main/java/com/iic/shopingo/ui/request_flow/activities/SelectShopperActivity.java

```diff
@@ -9,24 +10,40 @@
 import butterknife.ButterKnife;
 import butterknife.InjectView;
 import butterknife.OnItemClick;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.location.LocationServices;
 import com.iic.shopingo.R;
+import com.iic.shopingo.services.CurrentLocationProvider;
 import com.iic.shopingo.ui.request_flow.views.SelectShopperListItemView;
 import java.util.ArrayList;
 import java.util.List;
 
-public class SelectShopperActivity extends ActionBarActivity {
+public class SelectShopperActivity extends ActionBarActivity implements GoogleApiClient.ConnectionCallbacks {
+  private final long REQUEST_INTERVAL = 10 * 1000; // 10 seconds in millisecnods
+
   @InjectView(R.id.select_shopper_list)
   ListView shopperList;
 
   private SelectShopperAdapter adapter;
 
+  private CurrentLocationProvider locationProvider;
+
+  private GoogleApiClient googleApiClient;
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_select_shopper);
     ButterKnife.inject(this);
 
-    adapter = new SelectShopperAdapter(new ArrayList<SelectShopperAdapter.Shopper>());
+    googleApiClient = new GoogleApiClient.Builder(this)
```

Comment:
I can see how this would get annoying if we have multiple activities that need location.
Why not make the location provider deal with it?